### PR TITLE
added direct subseed -> digest method

### PR DESCRIPTION
**Pull request by paulhandy**
This is helpful when creating merkle trees

### CODE REVIEW

File: sign/src/iss.rs [hunk 1]
```diff
@@ -144,6 +144,37 @@ where
...
+    T: Copy + Clone + Sized,
+    C: Curl<T>,
+{
+    let length = security * KEY_LENGTH / HASH_LENGTH;
+    let mut key_state: [T; STATE_LENGTH] = [curl.rate()[0]; STATE_LENGTH];
+    let mut digest_state: [T; STATE_LENGTH] = [curl.rate()[0]; STATE_LENGTH];
+    curl.absorb(subseed);
+
+    key_state.clone_from_slice(curl.state());
+    for __ in 0..(length) {
```
Comment (th0br0) [inline sign/src/iss.rs]: why `__` and not plain `_`? it's a generic wildcard

File: sign/src/iss.rs [hunk 2]
```diff
@@ -144,6 +144,37 @@ where
...
+        curl.state_mut().clone_from_slice(&key_state);
+        curl.squeeze(out);
+        key_state.clone_from_slice(curl.state());
```
Comment (th0br0) [inline sign/src/iss.rs]: The copying just feels bad :/ but I guess it's the only way right now.

**Review by th0br0 — CHANGES_REQUESTED** [review]
You broke dem tests. Please fix? 
IIRC the inputs & outputs are equivalent to Java tests.

File: sign/src/iss.rs (hunk 2 above)
Comment (paulhandy) [reply sign/src/iss.rs]: The other alternative is to have absorb/squeeze accept a sponge context, which I think might be a wise way of doing it.

**Pull request merged**